src/components/ReportingConfig/ReportingConfigForm.jsx

```diff
@@ -0,0 +1,436 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import snakeCase from 'lodash/snakeCase';
+import isEmpty from 'lodash/isEmpty';
+import { ValidationFormGroup, Input, StatefulButton, Icon } from '@edx/paragon';
+
+//  All the fields in this form that need to be validated can be added here.
+const REQUIRED_FIELDS = [
+  'hourOfDay',
+];
+// for the email delivery method
+const REQUIRED_EMAIL_FIELDS = [
+  'email',
+];
+// for the sftp delivery mothod
+const REQUIRED_SFTP_FIELDS = [
+  'sftpPort',
+  'sftpHostname',
+  'sftpUsername',
+  'sftpPassword',
+  'sftpFilePath',
+];
+/**
+ * Validates this form. If the form is valid, it will return true.
+ * Otherwise, it returns the invalid fields.
+ * @param {FormData} formData
+ * @param {[String]} requiredFields
+ */
+const validateReportingForm = (formData, requiredFields) => {
+  const invalidFields = requiredFields
+    .filter(field => !formData.get(field))
+    .reduce((prevFields, currField) => ({ ...prevFields, [currField]: true }), {});
+  return isEmpty(invalidFields) || invalidFields;
+};
+
+/**
+ * Transform the form data from camelCase back into snake case to
+ * prepare it to be sent down the wire.
+ * @param {*} formData
+ */
+const transformFormData = (formData) => {
+  const transformedData = new FormData();
+  [...formData.entries()]
+    .forEach(entry => (
+      transformedData.append(snakeCase(entry[0]), entry[1])
+    ));
+  return transformedData;
+};
+
+/**
+ * Used as both the form to create and edit an existing reporting configuration.
+ * @param {Object} config -> incoming configuration, if it exists
+ */
+const ReportingConfigForm = ({
+  config,
+  createConfig,
+  updateConfig,
+}) => {
+  const [frequency, setFrequency] = useState(config ? config.frequency : 'monthly');
+  const [deliveryMethod, setDeliveryMethod] = useState(config ? config.deliveryMethod : 'email');
+  const [invalidFields, setInvalidField] = useState({});
+  const [active, setActive] = useState(config ? config.active : false);
+  const [submitState, setSubmitState] = useState('default');
+  return (
+    <form
+      onSubmit={async (e) => {
+        e.preventDefault();
+        setSubmitState('pending');
+        const formData = new FormData(e.target);
+        const requiredFields = REQUIRED_FIELDS;
+        if (deliveryMethod === 'email') {
+          requiredFields.push(...REQUIRED_EMAIL_FIELDS);
```

Comment:
Could you construct these as Constants above and then just set the required fields below?

```
const REQUIRED_FIELDS = [
  'hourOfDay',
];
// for the email delivery method
const REQUIRED_EMAIL_FIELDS = [
  ...REQUIRED_FIELDS,
  'email',
];
// for the sftp delivery mothod
const REQUIRED_SFTP_FIELDS = [
  ...REQUIRED_FIELDS,
  'sftpPort',
  'sftpHostname',
  'sftpUsername',
  'sftpPassword',
  'sftpFilePath',
];
```

Reply:
handled in the refactor